lib/hw_node.py

```diff
@@ -66,36 +83,111 @@ def get_salt_cmd(sls, node):
+            re.search("\d\d\d\d-\d\d-\d\d\s+\d\d:\d\d:\d\d\s*$" , str)):
+            return str
+    return 'no_meaningful_line_found'
+
+def cold_restart(mode):
+    pass
+
+def wait_node_is_ready(node,
+                        timeout=900,
+                        conman_line_max_age=None,
+                        max_cold_restart=3,
+                        port_lookup=22,
+                        port_lookup_timeout=None,
+                        port_lookup_attempts=None ):
+    """ Return sign if node booted or not in timeout
+        timeout=overall timeout for node boot and start ssh
+        (ssh starts after end of kiwi provisioning)
+    """
+    if port_lookup_timeout is None:
+        port_lookup_timeout = default_port_lookup_timeout
+    if port_lookup_attempts is None:
+        port_lookup_attempts = default_port_lookup_attempts
+    if conman_line_max_age is None:
+        conman_line_max_age =  default_conman_line_max_age
+
+    starttime = time.time()
+    conman_line = 'start_line'
+    conman_line_time = time.time()
+    conmanfile = conman_log_prefix + node['node'].split('.')[0]
+    cold_restart_count = 0
+
+    while starttime+timeout > time.time():
+
+        # check last line in conman log
+        new_conman_line = read_last_meaningful_line(conmanfile)
+        if conman_line != new_conman_line:
+            #logging.debug("New log detected:"+new_conman_line)
+            conman_line = new_conman_line
+            conman_line_time = time.time()
+        if (time.time() - conman_line_time) > conman_line_max_age:
+            logging.info("Node boot failure detected, make cold restart")
+            cold_restart(node)
+            cold_restart_count += 1
+            if cold_restart_count <= max_cold_restart:
+                logging.error(
+                "Achieved max cold restart couter %s for node %s,throws exception"
+                        % (max_cold_restart, node['node']))
+                raise CannotBootException("max cold restart couter (%s) for %s" %
+                         (max_cold_restart, node['node']))
+            next
+
+        #check port status
+        try:
+            logging.debug("timeout=" + str(port_lookup_timeout))
+            local = LocalNode()
+            local.wait_for_port(host=node['ip'],
+                            port = port_lookup,
+                            timeout=port_lookup_timeout,
+                            attempts=port_lookup_attempts)
+            logging.debug("Connected to node %s " % node['node'])
+            return True
+        except:
+            logging.error("Node {} have not started in timeout {}".format(
+                node['node'], port_lookup_timeout))
+
+    raise TimeoutException("{} have not started in timeout {}".format(
+        node['node'], timeout))
+
+def minimal_needed_configuration(node, timeout=60, add_sls=[]):
+    sls_list.extend(add_sls)
+    logging.debug('Executing salt script[{}]'.format(sls_list))
     for sls in sls_list:
```

Comment:
It is very strange to call the parameter `add_sls` when you don't want rename iterating variable :-D

Reply:
renamed